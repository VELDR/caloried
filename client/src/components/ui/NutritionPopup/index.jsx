import PropTypes from 'prop-types';
import CountUp from 'react-countup';
import { FormattedMessage } from 'react-intl';
import { Dialog, DialogContent, DialogTitle, Tooltip } from '@mui/material';
import proteinIcon from '@static/images/protein.png';
import fatIcon from '@static/images/fat.png';
import carbsIcon from '@static/images/carbs.png';

import classes from './style.module.scss';

const NutritionPopup = ({ open, onClose, user }) => (
  <Dialog className={classes.container} open={open} onClose={onClose} data-testid="nutrition-popup">
    <DialogTitle className={classes.header}>
      <div className={classes.header__title}>
        <FormattedMessage id="app_welcome" /> {user?.username}!
      </div>
      <div className={classes.header__subtitle}>
        <FormattedMessage id="app_your_personalized_plan" />
      </div>
    </DialogTitle>
    <DialogContent className={classes.content}>
      <div className={classes.message}>
        <div className={classes.message__content}>
          <FormattedMessage id="app_to" /> <span>{user?.goal}</span> <FormattedMessage id="app_weight_need_consume" />
        </div>
        <span className={classes.calories}>
          <CountUp end={user?.bmr} duration={2} />
        </span>
        <div className={classes.message__kcal}>
          kcal/
          <FormattedMessage id="app_day" />
        </div>
      </div>
      <div className={classes.message2}>
        <FormattedMessage id="app_your_daily_nutritional_goals" />
      </div>
      <div className={classes.macronutrient}>
        <div className={classes.macronutrient__item}>
          <Tooltip title="Protein">
            <img src={proteinIcon} alt="Protein" />
          </Tooltip>{' '}
          <CountUp end={user?.proteinIntake} duration={2} suffix="g" />
        </div>

        <div className={classes.macronutrient__item}>
          <Tooltip title="Carbohydrate">
            <img src={carbsIcon} alt="Carbs" />
          </Tooltip>{' '}
          <CountUp end={user?.carbsIntake} duration={2} suffix="g" />
        </div>

        <div className={classes.macronutrient__item}>
          <Tooltip title="Fat">
            <img src={fatIcon} alt="Fat" />
          </Tooltip>{' '}
          <CountUp end={user?.fatIntake} duration={2} suffix="g" />
        </div>
      </div>
      <div className={classes.button} onClick={onClose} data-testid="nutrition-popup-button">
        <FormattedMessage id="app_got_it" />
      </div>
    </DialogContent>
  </Dialog>
);

NutritionPopup.propTypes = {
  open: PropTypes.bool,
  onClose: PropTypes.func,
  user: PropTypes.object,
};

export default NutritionPopup;
