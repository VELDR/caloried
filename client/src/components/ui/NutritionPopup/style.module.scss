@import '@styles/base';

.container {
  text-align: center;
  .header {
    border-bottom: 5px solid var(--color-text-secondary);
    font-family: Poppins, 'sans-serif';
    &__title {
      font-size: $font-size-larger !important;
      font-weight: 700;
    }
    &__subtitle {
      font-size: $font-size-medium !important;
    }
  }

  .content {
    margin-top: 1rem;
    @include flex-center-column(1.5rem);
    .message {
      &__content {
        span {
          font-weight: 700;
        }
      }
      .calories {
        // background-color: var(--color-primary);
        // border-radius: 4rem;
        padding: 0 1rem;
        font-weight: 700;
        font-size: $font-size-huger;
      }
      &__kcal {
        font-weight: 300;
      }
    }
    .message2 {
      font-size: $font-size-large;
      font-weight: 500;
    }
    .macronutrient {
      @include flex-items-center(1rem);
      &__item {
        @include flex-items-center(0.5rem);
        img {
          @include box(2rem);
        }
      }
    }
    .button {
      @include primary-button;
      width: 100%;
    }
  }
}
