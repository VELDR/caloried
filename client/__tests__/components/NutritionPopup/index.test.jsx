import NutritionPopup from '@components/ui/NutritionPopup';
import { fireEvent, render } from '@utils/testHelper';

let isOpen = true;
let wrapper;

jest.mock('@static/images/protein.png', () => 'path/to/protein.png');
jest.mock('@static/images/carbs.png', () => 'path/to/carbs.png');
jest.mock('@static/images/fat.png', () => 'path/to/fat.png');

const mockProps = {
  open: isOpen,
  user: {
    username: 'dummy',
    goal: 'maintain',
    bmr: '1000',
    proteinIntake: '50',
    fatIntake: '50',
    carbsIntake: '50',
  },
  onClose: () => {
    isOpen = false;
  },
};

beforeEach(() => {
  wrapper = render(<NutritionPopup {...mockProps} />);
});

describe('NutritionPopup Component', () => {
  it('should render component correctly', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('nutrition-popup')).toBeInTheDocument();
  });

  it('should close popup', () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('nutrition-popup-button');
    fireEvent.click(button);
    expect(isOpen).toBe(false);
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
