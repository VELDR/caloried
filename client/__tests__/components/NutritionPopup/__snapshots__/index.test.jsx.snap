// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NutritionPopup Component should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      data-testid="nutrition-popup"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r8:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
            id=":r8:"
          >
            <div>
              Selamat Datang
               
              dummy
              !
            </div>
            <div>
              Rencana Nutrisi Pribadi Anda
            </div>
          </h2>
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            <div>
              <div>
                Untuk
                 
                <span>
                  maintain
                </span>
                 
                berat, perlu konsumsi
              </div>
              <span>
                <span>
                  0
                </span>
              </span>
              <div>
                kcal/
                hari
              </div>
            </div>
            <div>
              Target Nutrisi Harian Anda
            </div>
            <div>
              <div>
                <img
                  alt="Protein"
                  aria-label="Protein"
                  class=""
                  data-mui-internal-clone-element="true"
                  src="path/to/fat.png"
                />
                 
                <span>
                  0g
                </span>
              </div>
              <div>
                <img
                  alt="Carbs"
                  aria-label="Carbohydrate"
                  class=""
                  data-mui-internal-clone-element="true"
                  src="path/to/fat.png"
                />
                 
                <span>
                  0g
                </span>
              </div>
              <div>
                <img
                  alt="Fat"
                  aria-label="Fat"
                  class=""
                  data-mui-internal-clone-element="true"
                  src="path/to/fat.png"
                />
                 
                <span>
                  0g
                </span>
              </div>
            </div>
            <div
              data-testid="nutrition-popup-button"
            >
              Baik
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
